test/val/val_id_test.cpp

```diff
@@ -1925,6 +1925,169 @@ TEST_F(ValidateIdWithMessage, OpFunctionCallArgumentTypeBad) {
   CompileSuccessfully(spirv.c_str());
   EXPECT_EQ(SPV_ERROR_INVALID_ID, ValidateInstructions());
 }
+
+// Valid: OpSampledImage result <id> is used in the same block by
+// OpImageSampleImplictLod
+TEST_F(ValidateIdWithMessage, OpSampledImageGood) {
+  string spirv = kGLSL450MemoryModel + R"(
+       %void             = OpTypeVoid
```

Comment:
This is a lot of boilerplate to set up the types and function preamble.  Better to make everything before the first OpLabel its own module-scope constant character array, so that each test can focus on just its own special characteristics.

This will also mean you'll want to make nicer names for the %<number> IDs.